code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -503,6 +503,36 @@
 	new_frank.setGender(gender)
 	return new_frank
 
+//////////////////////////////////////////////
+//                                          //
+//               TIME AGENT                 //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/time_agent
+	name = "time agent anomaly"
+	role_category = /datum/role/time_agent
+	required_candidates = 1
+	weight = 4
+	cost = 10
+	requirements = list(70, 60, 50, 40, 30, 20, 10, 10, 10, 10)
+	logo = "time-logo"
+
+/datum/dynamic_ruleset/midround/from_ghosts/time_agent/acceptable(var/population=0,var/threat=0)
+	var/player_count = mode.living_players.len
+	var/antag_count = mode.living_antags.len
+	var/max_traitors = round(player_count / 10) + 1
+	if ((antag_count < max_traitors) && prob(mode.threat_level))
```

Comment:
` prob(mode.threat_level)` remove that. This ruleset already passed many checks before going to this point, no need to add yet another rng test.